packages/react-select-async-paginate/__stories__/AutoloadTest.stories.tsx

```diff
@@ -0,0 +1,113 @@
+import { within, userEvent, expect, waitFor, fireEvent } from "@storybook/test";
+import type { Meta, StoryObj } from "@storybook/react";
+import type { GroupBase } from "react-select";
+import { fn } from "@storybook/test";
+
+import { AsyncPaginate } from "../src";
+import type { LoadOptions } from "../src";
+
+import { Autoload, loadOptions } from "./Autoload";
+
+const meta: Meta<typeof Autoload> = {
+  title: "react-select-async-paginate/Autoload",
+  component: Autoload,
+};
+export default meta;
+type Story = StoryObj<typeof AsyncPaginate>;
+
+export const AutoloadTest: Story = {
+  name: "Interaction",
+  args: {
+    loadOptions: fn(
+      loadOptions as LoadOptions<unknown, GroupBase<unknown>, unknown>
+    ),
+  },
+  play: async ({ canvasElement, step, args }) => {
+    const canvas = within(canvasElement);
+    const mockLoadOptions = args.loadOptions;
+
+    await step("Autoloading after rendering a component", async () => {
+      await waitFor(() => {
+        expect(mockLoadOptions).toHaveBeenCalledTimes(1);
+      });
+    });
+
+    await step("Click on the Select to display the options list", async () => {
+      const select = canvas.getByRole("combobox");
+
+      await userEvent.click(select, {
+        delay: 400,
+      });
+
+      await waitFor(() => {
+        expect(canvas.getByRole("listbox")).toBeVisible();
+      });
+    });
+
+    await step(
+      "Scroll the options list to the end of first pagination page",
+      async () => {
+        await waitFor(() => {
+          const listbox = canvas.getByRole("listbox");
+          fireEvent.scroll(listbox, { target: { scrollTop: 500 } });
+
+          expect(canvas.getByText("Option 10")).toBeVisible();
+        });
+      }
+    );
+
+    await step(
+      "Scroll the options list to the end of second pagination page",
+      async () => {
+        await waitFor(
+          () => {
+            const listbox = canvas.getByRole("listbox");
+            fireEvent.scroll(listbox, { target: { scrollTop: 500 } });
+
+            expect(canvas.getByText("Option 20")).toBeVisible();
+          },
+          {
+            timeout: 4000,
+          }
+        );
+      }
+    );
+
+    await step("Type into the Select", async () => {
+      const select = canvas.getByRole("combobox");
+      const listbox = canvas.getByRole("listbox");
+
+      await waitFor(
+        async () => {
+          await userEvent.type(select, "Option 40", { delay: 100 });
```

Comment:
Looks like it should be moved outside of `waitFor`

And the good practice for tests to have `waitFor` contain only one `expect` check